logstash-core/src/test/java/org/logstash/ackedqueue/QueueTest.java

```diff
@@ -473,38 +472,37 @@ public void ackingMakesQueueNotFullAgainTest() throws IOException, InterruptedEx
 
         // allow 10 elements per page but only 100 events in total
         Settings settings = TestSettings.volatileQueueSettings(singleElementCapacity * 10, singleElementCapacity * 100);
-
-        TestQueue q = new TestQueue(settings);
-        q.open();
-
-        int ELEMENT_COUNT = 90; // should be able to write 90 events (9 pages) before getting full
-        for (int i = 0; i < ELEMENT_COUNT; i++) {
-            long seqNum = q.write(element);
-        }
-
-        assertThat(q.isFull(), is(false));
-
-        // we expect this next write call to block so let's wrap it in a Future
-        Callable<Long> write = () -> {
-            return q.write(element);
-        };
         ExecutorService executor = Executors.newFixedThreadPool(1);
-        Future<Long> future = executor.submit(write);
-        assertThat(future.isDone(), is(false));
-
-        while (!q.isFull()) { Thread.sleep(10); }
-        assertThat(q.isFull(), is(true));
-
-        Batch b = q.readBatch(10); // read 1 page (10 events)
-        b.close();  // purge 1 page
-
-        while (q.isFull()) { Thread.sleep(10); }
-        assertThat(q.isFull(), is(false));
-
-        // will not complete because write will not unblock until the page is purge with a batch close/acking.
-        assertThat(future.isDone(), is(false));
-
-        q.close();
+        try (TestQueue q = new TestQueue(settings)) {
+            q.open();
+            int ELEMENT_COUNT =
+                90; // should be able to write 90 events (9 pages) before getting full
+            for (int i = 0; i < ELEMENT_COUNT; i++) {
+                q.write(element);
+            }
+            assertThat(q.isFull(), is(false));
+            // we expect this next write call to block so let's wrap it in a Future
+            Callable<Long> write = () -> q.write(element);
+            final long seq = q.nextSeqNum();
+            Future<Long> future = executor.submit(write);
+            assertThat(future.isDone(), is(false));
+            while (!q.isFull()) {
+                Thread.sleep(10);
+            }
+            assertThat(q.isFull(), is(true));
+            try (Batch b = q.readBatch(10)) {
+                assertThat(b.size(), is(10));  // purge 1 page by closing b
+            }
+            Thread.sleep(10L);
+            while (q.isFull()) {
+                Thread.sleep(10);
+            }
+            assertThat(q.isFull(), is(false));
+            assertThat(future.get(), is(seq + 1));
+        } finally {
+            executor.shutdownNow();
+            executor.awaitTermination(2L, TimeUnit.MINUTES);
```

Comment:
@colinsurprenant actually since we have a timeout on the test we can just wait on the executor without timeout and it will throw if it doesn't join :) Sec fixing